example-ideal.txt

```diff
@@ -0,0 +1,62 @@
+# A Picobot program that can fill an empty room
+# States 1 and 2: move to the top left
+# State 1: move left
+If you are holding your first weapon, 
+and there is anything towards the Shire, 
+anything towards the Lonely Mountain, 
+nothing towards the Undying Lands, 
+and anything towards Mordor, 
+go towards the Undying Lands and ready your first weapon.
+
+If you are holding your first weapon,
+and there is anything towards the Shire, 
+anything towards the Lonely Mountain, 
+orcs towards the Undying Lands, 
+and anything towards Mordor, 
+go towards the Shire and ready your second weapon.
+
+# State 1: move up
+If you are holding your second weapon,
+and there is nothing towards the Shire,
+anything towards the Lonely Mountain,
+anything towards the Undying Lands,
+and anything towards Mordor,
+go towards the Shire and ready your second weapon.
+
+If you are holding your first weapon,
+and there are orcs towards the Shire,
+anything towards the Lonely Mountain,
+anything towards the Undying Lands,
+and nothing towards Mordor,
+go towards Mordor and ready your third weapon.
+
+# States 3 and 4: fill from top to bottom, left to right
+# State 3: fill this column to the bottom
+If you are holding your third weapon,
+and there is anything towards the Shire,
+anything towards the Lonely Mountain,
+anything towards the Undying Lands,
+and nothing towards Mordor,
+go towards Mordor and ready your third weapon.
+
+If you are holding your third weapon,
+and there is anything towards the Shire,
+nothing towards the Lonely Mountain,
+anything towards the Undying Lands,
+and orcs towards Mordor,
+go towards the Lonely Mountain and ready your fourth weapon.
+
+# State 4: fill this column to the top
+If you are holding your fourth weapon,
+and there is nothing towards the Shire,
+anything towards the Lonely Mountain,
+anything towards the Undying Lands,
+and anything towards Mordor,
+go towards the Shire and ready your fourth weapon.
+
+If you are holding your fourth weapon,
+and there are orcs towards the Shire,
+nothing towards the Lonely Mountain,
+anything towards the Undying Lands,
+and anything towards Mordor,
+go towards the Lonely Mountain and ready your third weapon.
```

Comment:
Agreed. I like the use of `orcs` as the indicator for the space being black. I think `towards` really adds to the silliness&mdash;I'm sad to see it go. :(